filament/src/materials/MaterialHelpersShading.fxh

```diff
@@ -0,0 +1,389 @@
+    if (materialParams.useNormalTexture == 1) {
+        // We combine the normals in world space, hence the transformation in the end from world to tangent, assuming an
+        // orthonormal tangent frame (which may not hold actually but looks fine enough for now).
+        vec3 normalWS = TriplanarNormalMap(materialParams_normalTexture,
+                                           materialParams.textureScaler.y,
+                                           fragmentData.pos,
+                                           fragmentData.normal);
+        material.normal = normalWS * getWorldTangentFrame();
+    }
+    // By this time the normal should be ready, it is safe to apply the normal scale
+    material.normal.xy *= materialParams.normalIntensity;
+#endif
+}
+
+void ApplyClearCoatNormalMap(inout MaterialInputs material, inout FragmentData fragmentData) {
+#if defined(MATERIAL_HAS_CLEAR_COAT_NORMAL)
+    if (materialParams.useClearCoatNormalTexture == 1) {
+        vec3 clearCoatNormalWS = TriplanarNormalMap(materialParams_clearCoatNormalTexture,
+                                                    materialParams.textureScaler.z,
+                                                    fragmentData.pos,
+                                                    fragmentData.normal);
+        material.clearCoatNormal = clearCoatNormalWS * getWorldTangentFrame();
+        material.clearCoatNormal.xy *= materialParams.clearCoatNormalIntensity;
+    }
+#endif
+}
+
+void ApplyBaseColor(inout MaterialInputs material, inout FragmentData fragmentData) {
+#if defined(MATERIAL_HAS_BASE_COLOR) 
+    if (materialParams.useBaseColorTexture == 1) {
+#if defined(BLENDING_ENABLED) || defined(HAS_REFRACTION)
+        material.baseColor.rgba = TriplanarTexture(materialParams_baseColorTexture,
+                                                   materialParams.textureScaler.x,
+                                                   fragmentData.pos,
+                                                   fragmentData.normal)
+                                      .rgba;
+#else
+        material.baseColor.rgb = TriplanarTexture(materialParams_baseColorTexture,
+                                                  materialParams.textureScaler.x,
+                                                  fragmentData.pos,
+                                                  fragmentData.normal)
+                                     .rgb;
+#endif
+    } else {
+#if defined(BLENDING_ENABLED) || defined(HAS_REFRACTION)
+        material.baseColor.rgba = materialParams.baseColor.rgba;
+#else
+        material.baseColor.rgb = materialParams.baseColor.rgb;
+#endif
+    }
+
+    // Naive multiplicative tinting seems to be fine enough for now
+    material.baseColor.rgb *= materialParams.tintColor.rgb;
+#if defined(BLENDING_ENABLED)
+    material.baseColor.rgb *= material.baseColor.a;
+#endif
+
+#if defined(BLENDING_ENABLED)
+    material.baseColor.a = 0.0;
+#else
+    material.baseColor.a = 1.0;
+#endif
```

Comment:
```suggestion
#if defined(BLENDING_ENABLED)
    material.baseColor.rgb *= material.baseColor.a;
    material.baseColor.a = 0.0;
#else
    material.baseColor.a = 1.0;
#endif
```

Reply:
Fixed in https://github.com/shapr3d/filament/pull/27/commits/f17aa20c12b5006d3662f3b03c47ac862af71a0d